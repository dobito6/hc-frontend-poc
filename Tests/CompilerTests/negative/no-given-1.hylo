trait P {}
trait Q {}

struct A {}
struct B {}

fun f<where A: Q>() {}

fun g() {
  given B: P {}
  given <where A: P> A: Q {}
  f()
}

fun h() {
  given <T where T: P> T: Q {}
  f()
}

fun i<T where T: P>(x: T) {
  i(x: ())
}
