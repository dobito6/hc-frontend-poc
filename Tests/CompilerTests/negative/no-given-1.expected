negative/no-given-1.hylo:12.3-4: error: no coercion from 'Q<A> ==> [Void]() let -> Void' to '[Void]() let -> Void'
  f()
  ^
negative/no-given-1.hylo:12.3-4: note: no given instance of 'Q<A>' in this scope
  f()
  ^
negative/no-given-1.hylo:17.3-4: error: no coercion from 'Q<A> ==> [Void]() let -> Void' to '[Void]() let -> Void'
  f()
  ^
negative/no-given-1.hylo:17.3-4: note: no given instance of 'Q<A>' in this scope
  f()
  ^
negative/no-given-1.hylo:21.3-4: error: no coercion from '<T> P<T> ==> [Void](x: let T) let -> Void' to '[Void](x: let Void) let -> Void'
  i(x: ())
  ^
negative/no-given-1.hylo:21.3-4: note: no given instance of 'P<Void>' in this scope
  i(x: ())
  ^
